src/main/java/com/example/provider/VehicleServlet.java

```diff
@@ -82,35 +88,54 @@ public VehicleServlet(
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     ServletUtils.setStandardResponseHeaders(response);
-
+    
     String vehicleId;
 
-    try {
-      vehicleId = ServletUtils.getEntityIdFromRequestPath(request);
-    } catch (IllegalArgumentException e) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "No Vehicle id provided.");
-      return;
-    }
+    vehicleId = ServletUtils.getEntityIdFromRequestPath(request);
 
-    logger.info(String.format("Getting vehicle with vehicleID: %s", vehicleId));
+    if (vehicleId != null && !vehicleId.isEmpty()) {
+      logger.info(String.format("Getting vehicle with vehicleID: %s", vehicleId));
 
-    GetVehicleRequest getVehicleRequest =
+      GetVehicleRequest getVehicleRequest =
         GetVehicleRequest.newBuilder().setName(VehicleUtils.getVehicleName(vehicleId)).build();
 
-    Vehicle vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
-    servletState.setVehicleId(vehicleId);
+      Vehicle vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
+      servletState.setVehicleId(vehicleId);
 
-    logger.info(String.format("Vehicle:\n%s", vehicle));
+      logger.info(String.format("Vehicle:\n%s", vehicle));
 
-    if (vehicle != null) {
-      if (tripMatcher.triggerMatching(vehicle, vehicleId)) {
-        vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
+      if (vehicle != null) {
+        if (tripMatcher.triggerMatching(vehicle, vehicleId)) {
+          vehicle = vehicleServiceClient.getVehicle(getVehicleRequest);
+        }
       }
-    }
 
-    response.getWriter().print(GsonProvider.get().toJson(vehicle));
-    response.getWriter().flush();
+      response.getWriter().print(GsonProvider.get().toJson(vehicle));
+      response.getWriter().flush();
+    } 
+    else {
+      ListVehiclesRequest.Builder listVehiclesRequestBuilder = ListVehiclesRequest.newBuilder()
+        .setParent(VehicleUtils.PROVIDER_NAME)
+        .setVehicleState(VehicleState.ONLINE)
+        .setPageSize(100);
+        // Default page size set to 100; Expeditor use case only needs 5-10 vehicles on average
+    
+      ListVehiclesRequest listVehiclesRequest = listVehiclesRequestBuilder.build();
+
+      logger.info(String.format("Getting list of all vehicles"));
+      ListVehiclesPagedResponse listVehiclesResponse = vehicleServiceClient.listVehicles(listVehiclesRequest);
+      
+      ArrayList<Vehicle> vehicleArrayList = new ArrayList<Vehicle>();
+      // Only loop through first page for now because Expeditor use case only needs 5-10 vehicles on average
+      for (Vehicle vehicle: listVehiclesResponse.getPage().getValues()) {
```

Comment:
Nit: Is it possible to use `listVehiclesResponse.getPage().getValues()` directly in line 134?